test/powershell/Language/Scripting/ParameterBinding.Tests.ps1

```diff
@@ -0,0 +1,538 @@
+            $rs = [system.management.automation.runspaces.runspacefactory]::CreateRunspace()
+            $rs.open()
+            $ps = [System.Management.Automation.PowerShell]::Create()
+            $ps.Runspace = $rs
+
+            try 
+            {        
+                [void] $ps.AddScript($f + "get-foo")
+                $asyncResult = $ps.BeginInvoke()
+                $ps.EndInvoke($asyncResult)
+            
+                $ps.Streams.Error.Count | Should Be 1 # the host does not implement it.
+                $ps.InvocationStateInfo.State | Should Be 'Completed'
+            } finally {
+                $ps.Dispose()
+                $rs.Dispose()
+            }
+        }
+
+        It 'Mandatory parameters used in interactive host' {
+            $th = New-TestHost
+            $rs = [system.management.automation.runspaces.runspacefactory]::Createrunspace($th)
+            $rs.open()
+            $ps = [System.Management.Automation.PowerShell]::Create()
+            $ps.Runspace = $rs
+            
+            try
+            {
+                $ps.AddScript($f + "get-foo").invoke()
+                $prompt = $th.ui.streams.prompt[0]
+                $prompt | should Not BeNullOrEmpty
+                $result = $prompt.split(":")
+                $result[0] | Should Match 'get-foo'
+                $result[-1] | should be 'a'                
+            } finally {
+                $rs.Close()
+                $rs.Dispose()
+                $ps.Dispose()
+            }
+        }
+    }
+
+    It 'Non positional parameters' {        
+        function get-foo
+        {
+            [CmdletBinding()]
+            param($a)
+            $a
+        }
+
+        get-foo a | Should Be a
+        $b = get-foo -a b | Should Be b
+    }
+
+    It 'Positional parameters when only one position specified: position = 1' {
+        
+        function get-foo
+        {
+            param([Parameter(position=1)] $a);
+            $a
+        }
+
+        get-foo b | Should Be b
+    }    
+
+    It 'Positional parameters when only position specified: position = 2' {
+       
+        function get-foo
+        {
+            param([Parameter(position=2)] $a);
+            $a
+        }
+
+        get-foo b | Should Be b
+    }
+
+    It 'Multiple positional parameters case 1' {
+       
+        function get-foo
+        {
+            param(
+            [Parameter(position=1)] $a,
+            [Parameter(position=2)] $b
+            )
+
+            $a
+            $b
+        }
+
+        (get-foo c d) -join ',' | Should Be 'c,d'
+
+        (get-foo -a c d) -join ',' | Should Be 'c,d'        
+
+        (get-foo -a c -b d) -join ',' | Should Be 'c,d'
+        
+        (get-foo -b d c) -join ',' | Should Be 'c,d'
+        
+        (get-foo c -b d) -join ',' | Should Be 'c,d'
+        
+    }
+
+    It 'Multiple positional parameters case 2' {
+        function get-foo
+        {
+            param(
```

Comment:
>param [](start = 12, length = 5)

could you add a comment here indicating that the parameters are purposefully out of order?